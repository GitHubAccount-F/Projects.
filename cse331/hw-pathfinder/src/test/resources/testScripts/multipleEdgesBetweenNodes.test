#Tests Dijkstra's algorithm when there is multiple edges with different costs
#between two nodes

#Creates the graphs
CreateGraph graph1
CreateGraph graph2

#Adds nodes to the graphs
AddNode graph1 one
AddNode graph1 two
AddNode graph1 three

AddNode graph2 one
AddNode graph2 two
AddNode graph2 three

#Adds multiple edges between each node
AddEdge graph1 one two 5
AddEdge graph1 one two 4
AddEdge graph1 two three 2
AddEdge graph1 two three 3

AddEdge graph2 one two 5
AddEdge graph2 one two 4
AddEdge graph2 one two 1
AddEdge graph2 two three 3
AddEdge graph2 two three 7
AddEdge graph2 two three 6


#Finds the paths with the lowest cost between two nodes:
FindPath graph1 one three

FindPath graph2 one three