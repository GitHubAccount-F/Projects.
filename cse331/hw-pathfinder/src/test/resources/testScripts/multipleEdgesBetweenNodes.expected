#Tests Dijkstra's algorithm when there is multiple edges with different costs
#between two nodes

#Creates the graphs
created graph graph1
created graph graph2

#Adds nodes to the graphs
added node one to graph1
added node two to graph1
added node three to graph1

added node one to graph2
added node two to graph2
added node three to graph2

#Adds multiple edges between each node
added edge 5.000 from one to two in graph1
added edge 4.000 from one to two in graph1
added edge 2.000 from two to three in graph1
added edge 3.000 from two to three in graph1

added edge 5.000 from one to two in graph2
added edge 4.000 from one to two in graph2
added edge 1.000 from one to two in graph2
added edge 3.000 from two to three in graph2
added edge 7.000 from two to three in graph2
added edge 6.000 from two to three in graph2


#Finds the paths with the lowest cost between two nodes:
path from one to three:
one to two with weight 4.000
two to three with weight 2.000
total cost: 6.000

path from one to three:
one to two with weight 1.000
two to three with weight 3.000
total cost: 4.000